utils/cq/d_logging_check.py

```diff
@@ -189,23 +209,107 @@ def check_quote(self, line):
 
     def check_return(self, line):
         """Check for one return character"""
-        expected_newlines = 1
+        max_newlines = 1
         code = line.raw()
         if any(map(code.startswith, PREFIXES_NNL)):
-            expected_newlines = 0
+            max_newlines = 0
+
+        if '"%s",' in code:
+            line.note('Use of %s at end of log-line, unable to check')
+            return
 
         count = code.count('\\n')
-        if count < expected_newlines:
-            parts = code.split('"')
-            if len(parts) == 3:
-                new_line = f'{parts[0]}"{parts[1]}\\n"{parts[2]}'
-                line.fix(new_line)
-                line.warning("Line does not contain newline (autofixable)")
-            else:
-                line.warning("Line does not contain newline")
-        elif count > expected_newlines:
+        # Some logging calls contain multiple new-line characters and that's OK, as long as one of
+        # them isn't at the end of a line.
+        if max_newlines == 0 and count > max_newlines:
             line.warning("Line contains too many newlines")
 
+    def check_df_rc_dot(self, line):
+        """Check there is no . after DF_RC"""
+        code = line.raw()
+        count = code.count('DF_RC')
+        if count == 0:
+            return
+        code = code.replace('DF_RC ', 'DF_RC')
+        if 'DF_RC".\\n"' not in code:
+            return
+        code = code.replace('DF_RC".\\n"', 'DF_RC"\\n"')
+        line.fix(code)
```

Comment:
Looks like this would convert
`" DF_RC ".\n"` to
`" DF_RC"\n"`
Where the space _**before**_ remains but the space _**after**_ is removed.

Also, you might consider `re.sub` here. (regular expression substitute). For example:
```
>>> import re
>>> re.sub(r'DF_RC *"\.\\n', 'DF_RC"\\n', 'plain string')
'plain string'
>>> re.sub(r'DF_RC *"\.\\n', 'DF_RC"\\n', 'some error " DF_RC ".\\n')
'some error " DF_RC"\n'
```

Reply:
Yes it would.  The c formatting is very messed up after using this script to apply fixes or corrections, there's an assumption (and I think it tells you to) that the user should run git-clang-format afterwards.

Reply:
But if we're already fixing it, shouldn't we fix it correctly? Or instead of fixing at all, just error?